### ISSUE TITLE
Feature widget is showing in drop down only default "Any" in Team Name 

### ISSUE THREAD

**Issue by lenalevy** [Hygieia/Hygieia#1850]
![configure feature widget](https://user-images.githubusercontent.com/33826321/33020791-2fdfccb4-ce08-11e7-86ef-7dafa7f16980.PNG)
Feature widget is showing in drop down only default "Any" in Team Name 
What should be change in order to present list of teams ?

**Comment by lenalevy** [comment]
Can somebody assist me ?

**Comment by subodhbattina** [comment]
Hi lenalevy: 
Please set this property "feature.jiraBoardAsTeam=true" in your application.properties file. And then also set the property "feature.jiraTeamFieldName" to any non-empty value.

And then rerun your jira collector to populate team data into your mongodb.
Clear your history from your browser, and bring your UI up and see if you are seeing the list of teams now. Thanks.

**Comment by lenalevy** [comment]
I did all the steps , but I still didn't get the list of teams

**Comment by lenalevy** [comment]
Could you please explain me from there exactly it took the list of teams in Jira. 
I'm  not able to fix the drop down of team list :(

**Comment by subodhbattina** [comment]
Hi @lenalevy,
I am truly sorry I couldn't follow up on this until now, but will look at this tomorrow and get back to you.
Thanks for your patience !

**Comment by subodhbattina** [comment]
Hi @lenalevy,

It may be that the "team" document (table) in your database does not have any team data.

1. Could you please check in your database and very if "team" has data in it?

2. Do you mind also going to the URL such as below, and see if you are getting any team data in response?
https://<your jira base url>/rest/agile/1.0/board?startAt=0

3. Also, when you run your JIRA collector, please check if there are any exceptions in the log, and share if you see any exceptions.

4. Finally, just to make sure, you have all these properties below configured in your properties file before running your JIRA collector.

feature.pageSize=100

// Adjust these values as per your needs
feature.deltaStartDate=2016-03-01T00:00:00.000000
feature.masterStartDate=2016-03-01T00:00:00.000000
feature.deltaCollectorItemStartDate=2016-03-01T00:00:00.000000

// Chron schedule: S M D M Y [Day of the Week]
feature.cron=0 * * * * *

// ST Query File Details - Required
feature.queryFolder=jiraapi-queries
feature.storyQuery=story
feature.epicQuery=epic

// JIRA URL, put in your values
feature.jiraProxyUrl=
feature.jiraProxyPort=

//  Put in your values
feature.jiraBaseUrl=
feature.jiraQueryEndpoint=rest/api/2/

// For basic authentication, requires username:password as string in base64
// This command will make this for you:  echo -n username:password | base64
 feature.jiraCredentials=bXdmMDUzOjA5U2hha3lhTXVuaTA5

feature.jiraIssueTypeNames=Story
feature.jiraSprintDataFieldName=customfield_10007
feature.jiraEpicIdFieldName=customfield_10003
feature.jiraStoryPointsFieldName=customfield_10004
feature.jiraTeamFieldName=customfield_11248
feature.jiraBoardAsTeam=true

Thanks !

**Comment by lenalevy** [comment]
Thank you for you detail answer!
I will check and back to you

**Comment by lenalevy** [comment]
Hi
Thanks again for detail answer
1. The team doesn't has data :( 
I have run the following command db.team.find()
2. I didn't understand how to get any team data with URL
with the following like i'm getting to tasks of "DE Victorious Scrum Board"  https://jira/secure/RapidBoard.jspa?rapidView=2340&projectKey=DE
What should I change in order to get any team data ?
3. from the log : 
2018-01-17 20:24:00,011 INFO  c.c.d.collector.CollectorTask - Team Data               0        0s
attach log
[nohup.zip](https://github.com/capitalone/Hygieia/files/1639429/nohup.zip)
4. feature.jiraTeamFieldName=customfield_11248
How I can find this id(for example : _11248) ?

Hope after your answer i will able to progress 
Thanks a lot , Lena

**Comment by lenalevy** [comment]
I have try one more thing but also , doesn't succeed 
in application.properties do the following change 
feature.jiraTeamFieldName=CROSS-4257

# Set this to true if you use boards as team
feature.jiraBoardAsTeam=true

and restart the api
 less nohup.out
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) [na:1.8.0_131]
2018-01-18 15:28:01,053 INFO  c.c.d.collector.CollectorTask - Team Data               0        1s
2018-01-18 15:28:01,054 DEBUG c.c.d.c.p.ProjectDataClientImpl - Constructing data collection for the feature widget, project-level data...
2018-01-18 15:28:02,721 ERROR c.c.d.client.DefaultJiraClient - No result was available from Jira unexpectedly - defaulting to blank response. The reason for this fault is the following:RestClientException{statusCode=Optional.of(403), errorCollections=[]}
2018-01-18 15:28:02,723 DEBUG c.c.d.client.DefaultJiraClient - Exception
com.atlassian.jira.rest.client.api.RestClientException: org.codehaus.jettison.json.JSONException: A JSONObject text must begin with '{' at character 11 of









<html>

<head>
    <title>Forbidden (403)</title>





<script type="text/javascript">var contextPath = '';</script>
<script>
window.WRM=window.WRM||{};window.WRM._unparsedData=window.WRM._unparsedData||{};window.WRM._unparsedErrors=window.WRM._unparsedErrors||{};
WRM._unparsedData["com.atlassian.plugins.atlassian-plugins-webresource-plugin:context-path.context-path"]="\"\"";
WRM._unparsedData["jira.webresources:feature-flags.feature-flag-data"]="{\"enabled-feature-keys\":[\"com.atlassian.jira.agile.darkfeature.editable.detailsview\",\"nps.survey.inline.dialog\",\"jira.plugin.devstatus.phasetwo\",\"jira.frother.reporter.field\",\"atlassian.rest.xsrf.legacy.enabled\",\"jira.issue.status.lozenge\",\"jira.project.config.new.version.menu\",\"com.atlassian.jira.config.BIG_PIPE\",\"com.atlassian.jira.projects.issuenavigator\",\"jira.sidebar.components.managecomponents\",\"com.atlassian.jira.config.PDL\",\"jira.plugin.devstatus.phasetwo.enabled\",\"jira.project.config.old.components.screen.disabled\",\"atlassian.aui.raphael.disabled\",\"app-switcher.new\",\"frother.assignee.field\",\"com.atlassian.jira.projects.ProjectCentricNavigation.Switch\",\"jira.onboarding.cyoa\",\"com.atlassian.jira.config.ProjectConfig.MENU\",\"com.atlassian.jira.projects.sidebar.DEFER_RESOURCES\",\"jira.zdu.admin-updates-ui\",\"jira.zdu.jmx-monitoring\",\"sd.new.settings.sidebar.location.disabled\",\"jira.sidebar.components.manageversions\",\"jira.zdu.cluster-upgrade-state\",\"com.atlassian.jira.config.CoreFeatures.LICENSE_ROLES_ENABLED\",\"com.atlassian.feedback.feedback-button-move-to-header-enable\",\"jira.export.csv.enabled\"],\"feature-flag-states\":{\"jira.jql.autoselectfirst\":true,\"jira.instrumentation.laas\":false,\"jira.create.linked.issue\":true}}";
WRM._unparsedData["jira.webresources:default-comment-security-level.DefaultCommentSecurityLevelHelpLink"]="{\"extraClasses\":\"default-comment-level-help\",\"title\":\"Commenting on an Issue\",\"url\":\"https://docs.atlassian.com/jira/jcore-docs-073/Editing+and+collaborating+on+issues#Editingandcollaboratingonissues-restrictacomment\",\"isLocal\":false}";

**Comment by lenalevy** [comment]
I had succeed to get team data with following URL
https://jira/rest/agile/1.0/board?startAt=0
I also succeed specific Scrum board that we are using
https://jira/rest/agile/1.0/board/2340
{"id":2340,"self":"https://jira/rest/agile/1.0/board/2340","name":"DE Victorious Scrum Board","type":"scrum

But again I don't understand which id i should put instead "XXXX"  in feature.jiraTeamFieldName=customfield_XXXX and how to find it

**Comment by lenalevy** [comment]
one more information that i found 
https://jira/rest/agile/1.0/board/1915/issue
I'm getting list of issues
BUT then i try https://jira/rest/agile/1.0/board/1915/project
I\m getting {"maxResults":50,"startAt":0,"total":0,"isLast":true,"values":[]} , and I have permissions

**Comment by lenalevy** [comment]
We don't have Tempo-Team plugin

**Comment by pversteijnen** [comment]
If you do not have the plugin you cannot use the team field. Then you want to use the feature.jiraBoardAsTeam=true

When true each board is seen as a team. When false the collector needs the Tempo-Team plugin.

**Comment by jagnathan** [comment]
I have set feature.jiraBoardAsTeam=true; I am able to see the boards in the Teams dropdown. I have also set feature.jiraTeamFieldName= as blank. So the collection Teams in DB has all the boards in it. In the feature collection, there are 1000+ issues, but the issues dont have any data in the Team section. Further even after selecting the project and  Team (actually board), no issues are shown in the graph. In the code, I dont see any connection between team(actually board) and story. In our project, the same issue could be in multiple boards.

My question is how is the relationship between board and story created inside Hygiea when there are no teams? In the code for Jira collector, the flag jiraBoardAsTeam is used only when populating Teams collection. See https://github.com/capitalone/Hygieia/pull/1644[https://github.com/capitalone/Hygieia/pull/1644](url)
